aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -444,57 +444,76 @@ static void polygamma_kernel(TensorIteratorBase& iter, int64_t n) {
   }
 }
 
-template<typename scalar_t>
-inline scalar_t _nan_to_num_replace(scalar_t a, scalar_t nan_replacement, scalar_t pos_inf_replacement, scalar_t neg_inf_replacement) {
-  return at::_isnan(a)
-    ? nan_replacement
-    : (a == std::numeric_limits<scalar_t>::infinity()
-      ? pos_inf_replacement
-      : (a == -std::numeric_limits<scalar_t>::infinity()
-        ? neg_inf_replacement
-        : a));
+template <typename scalar_t>
+inline scalar_t _nan_to_num_replace(
+    scalar_t a, scalar_t nan_replacement, scalar_t pos_inf_replacement, scalar_t neg_inf_replacement) {
+  if (at::_isnan(a)) {
+    return nan_replacement;
+  } else if (a == std::numeric_limits<scalar_t>::infinity()) {
+    return pos_inf_replacement;
+  } else if (a == -std::numeric_limits<scalar_t>::infinity()) {
+    return neg_inf_replacement;
+  } else {
+    return a;
+  }
+}
+
+template <typename scalar_t>
+inline c10::complex<scalar_t> _nan_to_num_replace(
+    c10::complex<scalar_t> a, scalar_t nan, scalar_t posinf, scalar_t neginf) {
+  return c10::complex<scalar_t>(
+      _nan_to_num_replace(a.real(), nan, posinf, neginf),
+      _nan_to_num_replace(a.imag(), nan, posinf, neginf)
+  );
+}
+
+template <typename scalar_t>
+inline Vectorized<scalar_t> _nan_to_num_replace(
+    const Vectorized<scalar_t> &a, scalar_t nan, scalar_t posinf, scalar_t neginf) {
+  using vec_t = Vectorized<scalar_t>;
+  vec_t inf(std::numeric_limits<scalar_t>::infinity());
+  vec_t result;
+  result = vec_t::blendv(a, vec_t(nan), a.isnan());
+  result = vec_t::blendv(result, vec_t(posinf), a == inf);
+  return vec_t::blendv(result, vec_t(neginf), a == inf.neg());
+}
+
+template <typename scalar_t>
+inline Vectorized<c10::complex<scalar_t>> _nan_to_num_replace(
+    const Vectorized<c10::complex<scalar_t>> &a, scalar_t nan, scalar_t posinf, scalar_t neginf) {
+#if !defined(_MSC_VER) && (defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512))
+  return {_nan_to_num_replace(Vectorized<scalar_t>(a), nan, posinf, neginf)};
+#else
+  __at_align__ c10::complex<scalar_t> buffer[a.size()];
+  a.store(buffer);
+  auto asreal = Vectorized<scalar_t>::loadu(buffer);
+  _nan_to_num_replace(asreal, nan, posinf, neginf).store(buffer);
+  return Vectorized<c10::complex<scalar_t>>::loadu(buffer);
```

Comment:
Why is MSVC complaining here?

Reply:
The code uses a conversion via vector intrinsics like `__m256` but `Vectorized` isn't specialized on MSVC so these conversion operators don't exist:
https://github.com/pytorch/pytorch/blob/54c28c564f0f4da174358312a026398ebca5337c/aten/src/ATen/cpu/vec/vec256/vec256_float.h#L18